### PR TITLE
[nomerge] Fix async creatable input

### PR DESCRIPTION
**Pull request by DanielHeath**
/cc @azaharakis

### CODE REVIEW

File: src/AsyncCreatable.js [hunk 1]
```diff
@@ -13,10 +13,6 @@ const AsyncCreatable = React.createClass({
 							<Select
 								{...asyncProps}
 								{...creatableProps}
-								onInputChange={(input) => {
-									creatableProps.onInputChange(input);
```
Comment (azaharakis) [inline src/AsyncCreatable.js]: Was that the purpose of calling it here, isn't createbleProps the onInputChange that the consumer passes in? Why doesn't it get called, is it being overridden by the Select component?

File: src/Creatable.js [hunk 2]
```diff
@@ -151,6 +151,14 @@ const Creatable = React.createClass({
 		});
 	},
 
+	componentWillUpdate (nextProps, nextState) {
+		// Tricky: Async clears the select inputValue when deleting
+		// an option, but it doesn't fire onInputChange.
```
Comment (azaharakis) [inline src/Creatable.js]: Is this comment relevant?

File: src/AsyncCreatable.js [hunk 3]
```diff
@@ -7,19 +7,42 @@ const AsyncCreatable = React.createClass({
...
+							const creatableRef = ref
+							return this.props.children({
+								...creatableProps,
+								ref: (select) => {
+									creatableRef(select)
+									asyncRef(select)
+									this.select = select
+								}
+							})
+						}}
```
Comment (azaharakis) [inline src/AsyncCreatable.js:23]: I'm still struggling a bit with this. What's the use case for needing to assign refs at each component layer? It doesn't really seem scalable?

Comment (DanielHeath) [reply src/AsyncCreatable.js:23]: https://github.com/realestate-com-au/react-select/blob/535d1b6ef1a7d62ad50dd84f0165c77bcf10d293/src/Async.js#L173 keeps a ref to the `Select`, as does https://github.com/realestate-com-au/react-select/blob/535d1b6ef1a7d62ad50dd84f0165c77bcf10d293/src/Creatable.js#L209

File: src/Creatable.js (hunk 2 above)
Comment (DanielHeath) [reply src/Creatable.js]: It matches the rest of the file - wherever something is done to make it work nicely with Async there's a comment like this.

Comment (azaharakis) [reply src/Creatable.js]: Right, but I guess I don't understand what the comment has to do with this new bit of code? Why does not firing onInputChange matter to setting the inputValue?

Comment (DanielHeath) [reply src/Creatable.js]: Ahh - I'll update the comment. The other place we keep `this.inputValue` in-sync is in an onInputChange handler.

File: src/AsyncCreatable.js (hunk 3 above)
Comment (azaharakis) [reply src/AsyncCreatable.js:23]: Cool, think I'm following now. 
Somewhat unrelated do we want to allow consumer to be able to assign its own child, shouldn't the last child always be a <Select /> component?

Comment (DanielHeath) [reply src/AsyncCreatable.js:23]: There are a bunch of cases where it's handy to be able to inject a child - e.g. testing, adding custom behavior that isn't natively present. I've taken the pattern from the implementation in `Async`.

**Comment by DanielHeath** [comment]
https://github.com/JedWatson/react-select/pull/1360 opened

**Pull request closed without merging**